docs/src/reference/compilation.rst

```diff
@@ -316,6 +316,128 @@ customize your build.
+    Mostly for reducing noise in Valgrind, only executes at process exit
+    (when all memory will be reclaimed anyways).
+    Default is 0.
+
+``annotate`` (True / False)
+    Generate an annotated HTML version of the input source files.
+    This has the same effect as the ``annotate`` argument in :func:`cythonize`.
+    Default is False.
+
+``fast_fail`` (True / False)
+    This will abort the compilation on the first error occurred rather than trying
+    to keep going and printing further error messages.
+    Default is False.
+
+``warning_errors`` (True / False)
+    Make all warnings into errors.
+    Default is False.
+
+``error_on_unknown_names`` (True / False)
+    Make unknown names an error.  Python raises a NameError when
+    encountering unknown names at runtime, whereas this option makes
+    them a compile time error.  If you want full Python compatibility,
+    you should disable this option and also 'cache_builtins'.
+    Default is True.
+
+``error_on_uninitialized`` (True / False)
+    Make uninitialized local variable reference a compile time error.
+    Python raises UnboundLocalError at runtime, whereas this option makes
+    them a compile time error. Note that this option affects only variables
+    of "python object" type.
+    Default is True.
+
+``convert_range`` (True / False)
+    This will convert statements of the form ``for i in range(...)``
+    to ``for i from ...`` when ``i`` is a C integer type, and the direction
+    (i.e. sign of step) can be determined.
+    WARNING: This may change the semantics if the range causes assignment to
+    i to overflow. Specifically, if this option is set, an error will be
+    raised before the loop is entered, whereas without this option the loop
+    will execute until an overflowing value is encountered.
+    Default is True.
+
+``cache_builtins`` (True / False)
+    Perform lookups on builtin names only once, at module initialisation
+    time.  This will prevent the module from getting imported if a
+    builtin name that it uses cannot be found during initialisation.
+    Default is True.
+
+``gcc_branch_hints`` (True / False)
+    Generate branch prediction hints to speed up error handling etc.
+    Default is True.
+
+``lookup_module_cpdef`` (True / False)
+    Enable this to allow one to write ``your_module.foo = ...`` to overwrite the
+    definition if the cpdef function foo, at the cost of an extra dictionary
+    lookup on every call.
+    If this is false it generates only the Python wrapper and no override check.
+    Default is False.
+
+``embed`` (True / False)
+    Whether or not to embed the Python interpreter, for use in making a
+    standalone executable or calling from external libraries.
+    This will provide a method which initialises the interpreter and
+    executes the body of this module.
+    See `this demo <https://github.com/cython/cython/tree/master/Demos/embed>`_
+    for a concrete example.
```

Comment:
"If true, the initialisation function is the C ``main()`` function, but this option can also be set to a non-empty string to provide a function name explicitly."